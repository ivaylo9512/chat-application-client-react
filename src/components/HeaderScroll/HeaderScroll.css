.header-scroll {
    height: 11%;
    overflow: hidden;
    transition: all 3s;

}
.header-scroll.hidden{
    margin-top: -11vh;
}
.header-scroll > div {
    height: 100%;
    width: 100%;
    display: flex;
    overflow-y: hidden;
}
.chat-info {
    color: white;
    width: 100%;
    text-align: center;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 2.5vh;
}
.chat-info span{
    border-radius: 1vw;
    padding: 0 3%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    font-weight: bold;
    font-size: 0.85vw;
    position: relative;
    background: rgb(24,27, 33);
    box-shadow: 0px 0px 0.6vh 0.2vh rgb(0 0 0 / 75%);
}