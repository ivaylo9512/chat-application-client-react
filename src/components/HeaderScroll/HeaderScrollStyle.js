import styled from 'styled-components'

export const Scroll = styled.div`
    height: 11%;
    overflow: hidden;
    transition: all 3s;

    margin-top: ${props => props.isHidden ? '-11vh' : '0'}
`
export const Container = styled.div`
    height: 100%;
    width: 100%;
    padding: 0 0.5%;
    display: flex;
    overflow-y: hidden;
    align-items: center;
`