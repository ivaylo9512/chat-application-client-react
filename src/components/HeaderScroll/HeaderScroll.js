import React, { useEffect, useState, useRef } from 'react';
import smoothscroll from 'smoothscroll-polyfill';
import './HeaderScroll.css'
import { getStylesHeaderState } from '../../app/slices/stylesSlice';

const HeaderScroll = () => {
    const listContainer = useRef()
    const scrollContainer = useRef()
    const [listPadding, setListPadding] = useState({})
    const name = useSelector(getChatSearchName)
    const isHidden = useSelector(getStylesHeaderState) 

    useEffect(() => {
        hideScrollBar()
    }, [name])
    
    useEffect(() => {
        window.addEventListener('resize', hideScrollBar)
        smoothscroll.polyfill()

        return () => window.removeEventListener('resize', hideScrollBar)
    }, []) 

    const hideScrollBar = () => {
        const height = parseFloat(window.getComputedStyle(listContainer.current).height)
        const containerHeight = parseFloat(window.getComputedStyle(scrollContainer.current).height)

        const barHeight = containerHeight - height
    
        setListPadding({paddingBottom:`${barHeight}px`})
    }

    const scroll = (e) => {
        e.currentTarget.scroll({left: e.currentTarget.scrollLeft + e.deltaY * 4 , top: e.currentTarget.scrollTop, behavior: 'smooth'})
    }

    return (
        <div className={`header-scroll${isHidden ? ' hidden' : ''}`} ref={scrollContainer}>
            <div ref={listContainer} style={listPadding} onWheel={scroll}>
                <ChatList />
            </div>
        </div>
    )
}

export default HeaderScroll