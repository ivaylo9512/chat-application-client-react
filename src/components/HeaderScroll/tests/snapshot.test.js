import * as redux from 'react-redux';
import HeaderScroll from 'components/HeaderScroll/HeaderScroll';
import { shallow } from 'enzyme';

describe('HeaderScroll snapshot tests', () => {
    beforeEach(() => {
        jest.spyOn(redux, 'useDispatch').mockReturnValue(jest.fn());
    }) 

    const createWrapper = (hiddenState, chatsState) => {
        jest.spyOn(redux, 'useSelector').mockReturnValueOnce(hiddenState)
            .mockReturnValueOnce(chatsState);
            
        return shallow(
            <HeaderScroll />
        )
    }

    it('should match snapshot', () => {
        const wrapper = createWrapper(true, { isLoading: false, data: { isLastPage: true }});

        expect(wrapper).toMatchSnapshot();
    })

    it('should match snapshot', () => {
        const wrapper = createWrapper(false, { isLoading: false, data: { isLastPage: true }});

        expect(wrapper).toMatchSnapshot();
    })
})