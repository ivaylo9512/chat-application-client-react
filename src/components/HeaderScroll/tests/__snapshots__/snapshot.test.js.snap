// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderScroll snapshot tests should match snapshot 1`] = `
<styled.div
  isHidden={true}
>
  <styled.div
    onWheel={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <ChatList />
  </styled.div>
</styled.div>
`;

exports[`HeaderScroll snapshot tests should match snapshot 2`] = `
<styled.div
  isHidden={false}
>
  <styled.div
    onWheel={[Function]}
    style={
      Object {
        "paddingBottom": 0,
      }
    }
  >
    <ChatList />
  </styled.div>
</styled.div>
`;
